source/aws_s3/s3.go

```diff
@@ -20,35 +20,43 @@ func init() {
 }
 
 type s3Driver struct {
-	s3client   s3iface.S3API
+	s3client   S3Client
 	config     *Config
 	migrations *source.Migrations
+	ctx        context.Context
 }
 
 type Config struct {
 	Bucket string
 	Prefix string
 }
 
+type S3Client interface {
+	ListObjects(ctx context.Context, params *s3.ListObjectsInput, optFns ...func(*s3.Options)) (*s3.ListObjectsOutput, error)
+	GetObject(ctx context.Context, params *s3.GetObjectInput, optFns ...func(*s3.Options)) (*s3.GetObjectOutput, error)
+}
+
 func (s *s3Driver) Open(folder string) (source.Driver, error) {
 	config, err := parseURI(folder)
 	if err != nil {
 		return nil, err
 	}
 
-	sess, err := session.NewSession()
+	ctx := context.Background()
+	cfg, err := awsconfig.LoadDefaultConfig(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	return WithInstance(s3.New(sess), config)
+	return WithInstance(ctx, s3.NewFromConfig(cfg), config)
 }
 
-func WithInstance(s3client s3iface.S3API, config *Config) (source.Driver, error) {
+func WithInstance(ctx context.Context, s3client S3Client, config *Config) (source.Driver, error) {
 	driver := &s3Driver{
 		config:     config,
 		s3client:   s3client,
 		migrations: source.NewMigrations(),
+		ctx:        ctx,
```

Comment:
Better off not storing ctx in s3Driver, instead pass it directly to loadMigrations

Reply:
Thank you for your feedback, Philip! I've updated the sources accordingly.